quickwit/quickwit-storage/src/lib.rs

```diff
@@ -383,9 +383,26 @@ pub(crate) mod test_suite {
     #[cfg(feature = "integration-testsuite")]
     pub async fn storage_test_multi_part_upload(storage: &mut dyn Storage) -> anyhow::Result<()> {
         let test_path = Path::new("hello_large.txt");
-        let test_buffer = vec![0u8; 15_000_000];
+
+        let mut test_buffer = Vec::with_capacity(15_000_000);
+        for i in 0..15_000_000u32 {
+            test_buffer.push((i % 256) as u8);
+        }
+
+        let expected_hash = md5::compute(&test_buffer);
```

Comment:
nit: I don't think we need to compute an MD5 digest, we can just clone `test_buffer` to keep it around and do a byte-compare  with the downloaded data.